pkg/operator/starter.go

```diff
@@ -160,7 +154,10 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 	if err != nil {
 		return err
 	}
-	migrator := migrators.NewInProcessMigrator(dynamicClientForMigration, kubeClient.Discovery())
+
+	migrationClient := kubemigratorclient.NewForConfigOrDie(controllerContext.KubeConfig)
+	migrationInformer := migrationv1alpha1informer.NewSharedInformerFactory(migrationClient, time.Minute*30)
```

Comment:
do we have to wait until `migrationInformer` has synced?

Reply:
I think yes, but I am not entirely sure(still learning and confused about informers and clients)

Reply:
`KubeStorageVersionMigrator.AddEventHandler`  does that already, adding the `HasSynced` func to the list of sync funcs of the migration controller. Should be fine.